provider/azure/internal/azurecli/az.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package azurecli
+
+import (
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+)
+
+// Logger for the Azure provider.
+var logger = loggo.GetLogger("juju.provider.azure.internal.azurecli")
+
+// executor is the subset of the exec.Command functionality used by Exec.
+type executor interface {
+	Output() ([]byte, error)
+}
+
+// newExecutor is used to create a new exec.Command for Exec. It is a
+// variable so that it can be overridden in tests.
+var newExecutor = func(command string, args []string) executor {
```

Comment:
This could perhaps be a field in a struct with methods GetAccessToken, ShowAccount, etc.
Then external code could mock it out if desired and you get a type that implements
the interface suggested in azure/environprovider.go.